src/cmd/run.go

```diff
@@ -58,20 +57,17 @@ func doRun(cmd *cobra.Command, args []string) {
 
 		pkg.StartMetricsServer(string(runner.Id), viper.GetInt("metrics-port"))
 
-		if viper.GetBool("scaling-enabled") {
-			config, err := pkg.GetKubernetesConfig()
-			pkg.CheckErr(err)
-
-			k8sClient := clientset.NewForConfigOrDie(config)
+		ctx := signal.Init(context.Background())
 
-			log.Info().Msgf("electing leader...")
-			go electLeader(k8sClient, runner.Id)
+		if viper.GetBool("scaling-enabled") {
+			leaseLockName := viper.GetString("runner-deployment")
+			leaseLockNamespace := viper.GetString("runner-pod-namespace")
+			lockIdentity := viper.GetString("runner-pod-name")
+			cobra.CheckErr(pkg.RunLeaderElection(ctx, runner.Id, leaseLockName, lockIdentity, leaseLockNamespace))
 		}
 
-		stop := pkg.InitSignalHandler()
-		wg := startWorkers(runner.Id, stop)
-		<-stop // Enter Forever Loop
-		log.Info().Msgf("interupt - waiting for jobs to complete ...")
+		wg := startWorkers(ctx, runner.Id)
+		time.Sleep(1 * time.Second)
```

Comment:
I carried over the "sleep" from opslevel-agent as IIRC it takes a second or 2 for the workers to start up and thus the `wg.Wait` wouldn't actually wait.   Theres probably some bug here i need to understand better.